rules/share-your-developer-secrets-securely/rule.md

```diff
@@ -0,0 +1,112 @@
+---
+type: rule
+archivedreason:
+title: Do you share your developer secrets securely?
+guid: 6165deaa-a3b0-40b1-8659-482a34faf7c4
+uri: share-your-developer-secrets-securely
+created: 2023-08-14T00:00:00.0000000Z
+authors:
+  - title: Gordon Beeming
+    url: https://ssw.com.au/people/gordon-beeming
+related: 
+  - store-your-secrets-securely
+redirects: []
+---
+
+Most systems will have variables that need to be stored securely; OpenId shared secret keys, connection strings, and API tokens to name a few.
+
+These secrets **must not** be stored in source control. It is insecure and means they are sitting out in the open, wherever code has been downloaded, for anyone to see.
+
+[Do you store your secrets securely?](https://www.ssw.com.au/rules/store-your-secrets-securely/) shows different ways to store your secrets securely. When you use .NET User Secrets, you can store your secrets in a JSON file on your local machine. This is great for development, but how do you share those secrets securely with other developers in your organization?
+
+<!--endintro-->
+
+### Bad Examples
+
+::: greybox
+
+#### Do not store secrets in appsettings.Development.json
+
+The `appsettings.Development.json` file is meant for storing development settings. It is not meant for storing secrets. This is a bad practice because it means that the secrets are stored in source control, which is insecure.
+
+![](development-json.jpg)
+
+:::
+::: bad
+Figure: Bad practice - Overall rating: 1/10
+:::
+
+#### Sharing secrets via email/teams
+
+Sending secrets over teams is a terrible idea, the messages can land up in logs, but they are also stored in the chat history. Developers can delete the messages once copied out, although this extra admin adds friction to the process and is often forgotten.
+
+**Note:** Sending the secrets in email, adds even more admin for deleting the message and is probably the least secure way of transferring secrets.
```

Comment:
It's not just about the admin. See: https://www.linkedin.com/pulse/email-equivalent-electronic-postcard-really-lesley-cooley/

Reply:
Although this may not matter so much within an organisation as between organizations. But it's a dangerous precedent.

Reply:
I played it down to just 'admin'. Do you think we should add a don't use email section specifically and mention why it's even less secure than even Microsoft Teams?

Reply:
I've updated the blurb for now to mention it being less secure and the admin part is to remove some of the trace